### PR TITLE
ENG-10651: Reset DR applied tracker after command log replaying when r…

### PR DESCRIPTION
**Pull request by slowsoul**
…recovering an old replica cluster to a new normal cluster

**Problem and solution:**
if the cluster is recovering and has completed replaying command log, check if it is recovered as a producer cluster for active-passive DR. If it is, the DR applied trackers in sites should be reset and cleared as they are no longer valid.

Added a new type of task in ExecuteTask system procedure, RESET_DR_APPLIED_TRACKER, which will clear the DR applied trackers on every site.

### CODE REVIEW

File: src/frontend/org/voltdb/RestoreAgent.java [hunk 1]
```diff
@@ -1294,6 +1296,24 @@ public void onReplayCompletion() {
...
+        if (m_isLeader && m_action.doesRecover()) {
+            VoltDBInterface instance = VoltDB.instance();
+            if (!instance.getCatalogContext().database.getIsactiveactivedred() &&
+                instance.getReplicationRole() == ReplicationRole.NONE) {
+                ByteBuffer params = ByteBuffer.allocate(4);
+                params.putInt(ExecutionEngine.TaskType.RESET_DR_APPLIED_TRACKER.ordinal());
+                try {
+                    instance.getClientInterface().callExecuteTask(30000, params.array());
+                } catch (IOException | InterruptedException e) {
+                    e.printStackTrace();
```
Comment (luyangco) [inline src/frontend/org/voltdb/RestoreAgent.java]: Worth to put a warning here says reset DR applied tracker timeout

**Comment by luyangco** [comment]
Review complete.

File: src/frontend/org/voltdb/RestoreAgent.java (hunk 1 above)
Comment (slowsoul) [inline src/frontend/org/voltdb/RestoreAgent.java]: Fixed

File: src/frontend/org/voltdb/jni/ExecutionEngine.java [hunk 2]
```diff
@@ -63,7 +63,8 @@
         SET_DR_PROTOCOL_VERSION(3),
         SP_JAVA_GET_DRID_TRACKER(4),
         SET_DRID_TRACKER(5),
-        GENERATE_DR_EVENT(6);
+        GENERATE_DR_EVENT(6),
+        RESET_DR_APPLIED_TRACKER(7);
```
Comment (nshi) [inline src/frontend/org/voltdb/jni/ExecutionEngine.java]: I think these are defined in the EE as well. They should be synced with each other even though some of them are only implemented in Java.

Comment (slowsoul) [inline src/frontend/org/voltdb/jni/ExecutionEngine.java]: Yes indeed, I have made a pull request for this, https://github.com/VoltDB/voltdb/pull/3732

Comment (luyangco) [inline src/frontend/org/voltdb/jni/ExecutionEngine.java]: Both branches were merged.

**Pull request merged**